rust/lon/src/lock/v1.rs

```diff
@@ -55,3 +56,17 @@ pub struct GitHubSource {
     pub url: String,
     pub hash: NixHash,
 }
+
+#[derive(Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct TarballSource {
+    pub fetch_type: FetchType,
+    #[serde(default, skip_serializing_if = "std::ops::Not::not")]
+    pub frozen: bool,
+
+    pub origin: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub revision: Option<String>,
```

Comment:
Do you want to also implement `lastModified`? Does Forgejo return lastModified?

Reply:
According to the forgejo source, they do not give it

```go
	ctx.Resp.Header().Add("Link", fmt.Sprintf("<%s/archive/%s.tar.gz?rev=%s>; rel=\"immutable\"",...)
```